client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -133,8 +149,17 @@ const Checkout = React.createClass( {
 			return `/plans/${ this.props.sites.getSelectedSite().slug }/thank-you`;
 		}
 
-		// TODO: include the receipt ID in this URL when it is present
-		return `/checkout/${ this.props.sites.getSelectedSite().slug }/thank-you`;
+		if ( this.props.transaction.step.data && this.props.transaction.step.data.receipt_id ) {
+			receiptId = this.props.transaction.step.data.receipt_id;
+
+			this.props.fetchReceiptCompleted( receiptId, {
+				receiptId: receiptId,
+				purchases: this.getPurchasesFromReceipt()
+			} );
+		}
```

Comment:
I'm not a big fan of accessing the transaction data directly like this. It would be great to hide that logic by using the receipts reducer. To be more specific, what I suggest is to update the `store-transactions` module to update the [receipts state tree](https://github.com/Automattic/wp-calypso/tree/master/client/state/receipts) when [receiving a response](https://github.com/Automattic/wp-calypso/blob/master/client/lib/store-transactions/index.js#L172) from the API. What do you think? That way we would also reduce the number of lines of code by reusing the [receipt assembler](https://github.com/Automattic/wp-calypso/blob/master/client/state/receipts/assembler.js).